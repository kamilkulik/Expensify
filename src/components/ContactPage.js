import React from 'react';

const ContactPage = () => (
    <div>
        This is my contact page
    </div>
);

export default ContactPage;