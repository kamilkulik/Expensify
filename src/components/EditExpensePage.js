import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (id, expense) => {
    this.props.editExpense(id, expense);
    this.props.history.push('/');
  }
  onClick = ({ id }) => {
    this.props.removeExpense( { id } );
    this.props.history.push('/');
  }
  render() {
    const id = this.props.match.params.id;
    return (
      <div>
        Editing the expense with id of {this.props.match.params.id}:
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />
        <button 
          onClick={this.onClick}
        >
        Remove
        </button>
      </div>
    );
  }
  
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id, expense)),
    removeExpense: (data) => dispatch(removeExpense( data ))
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
