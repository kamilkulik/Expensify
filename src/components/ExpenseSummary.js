import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../utils/expensesTotal';
import numeral from 'numeral';

export class ExpenseSummary extends React.Component {
  render() {
    return (
      <div>
        {`Showing ${this.props.expenses.length} 
        expense${this.props.expenses.length === 1 ? '' : 's'} 
        totalling ${numeral(expensesTotal(this.props.expenses) / 100).format('$0,0.00')}`}
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseSummary);