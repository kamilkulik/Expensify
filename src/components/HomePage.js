import React from 'react'

const Home = () => (
    <div>
        <h1>Welcome</h1>
        <p>Thanks for stopping by. Take a look around!</p>
    </div>
);

export default Home;