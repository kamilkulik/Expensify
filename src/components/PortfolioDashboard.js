import React from 'react';
import { NavLink } from 'react-router-dom';

const portfolioPages = [
    {
        item: 'First portfolio item',
    },
    {
        item: 'Second portfolio item',
    },
    {
        item: 'Third portfolio item',
    },
]

const PortfolioDashboard = () => {
    return (
    <div>
        <h1>Checkout some of my work</h1>
        {portfolioPages.map((page, index) => {
            return (
                <NavLink key={index} to={`/work/${index+1}`} activeClassName="is-active">{page.item}</NavLink>
            )
        })}
    </div>
    )
};

export default PortfolioDashboard;