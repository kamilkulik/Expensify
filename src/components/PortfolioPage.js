import React from 'react';

const PortfolioPage = ({match, location}) => {
    return (
        <div>
        <div>
            <p>
                <strong>Match props:</strong>
                <code>{JSON.stringify(match, null, 2)}</code>
            </p>
            <p>
            <strong>Location props:</strong>
            <code>{JSON.stringify(location, null, 2)}</code>
            </p>
        </div>
        <div>

            <h3>This is my first portfolio page</h3>
        </div>
        </div>
    )};

export default PortfolioPage;