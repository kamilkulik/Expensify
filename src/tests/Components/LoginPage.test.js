import React from 'react';
import { shallow } from 'enzyme'
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
  const loginSpy = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={loginSpy} />);
  wrapper.find('button').simulate('click');
  expect(loginSpy).toHaveBeenCalled();
});