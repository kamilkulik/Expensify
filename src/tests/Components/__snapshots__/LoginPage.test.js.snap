// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<div>
  <h1>
    Login to continue
  </h1>
  <button>
    Login
  </button>
</div>
`;
